### PR TITLE
Chapter 7: Sign Up

### PR DESCRIPTION
**Pull request by ainulfarizah**
ready

### CODE REVIEW

**Comment by ainulfarizah** [comment]
ready

File: app/views/new.html.erb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
Comment (huongnt-2545) [inline app/views/new.html.erb]: why you have this file???

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      <%= f.label      :name %>
+      <%= f.text_field :name %>
+      <%= f.label :email %>
+      <%= f.email_field :email %>
+      <%= f.label :password %>
+      <%= f.password_field :password %>
+      <%= f.label :password_confirmation, t("confirmation") %>
+      <%= f.password_field :password_confirmation %>
+      <%= f.submit t("create_my_account"), class: "btn btn-primary" %>
+    <%%= end >
```
Comment (huongnt-2545) [inline app/views/users/new.html.erb]:
```suggestion
    <% end %>
```

File: config/routes.rb [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
-  end
-end
+  root 'static_pages#home'
+  get 'static_pages/help'
 
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  get "/signup",  to: "users#new"
+  post "/signup", to:"users#create"
+  resources :users, only: %i(new create show)
+  end
```
Comment (huongnt-2545) [inline config/routes.rb]: add `end` close  keyword at the end of file

File: config/routes.rb [hunk 4]
```diff
@@ -1,9 +1,11 @@
 Rails.application.routes.draw do
+
+  get 'users/new'
```
Comment (huongnt-2545) [inline config/routes.rb]: remove this line

File: config/routes.rb [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
   scope "(:locale)", locale: /#{I18n.available_locales.join("|")}/ do
-    root 'static_pages#home'
-    get 'static_pages/help'
-    resources :users
-  end
-end
+  root 'static_pages#home'
```
Comment (huongnt-2545) [inline config/routes.rb:3]: indent = 2 spaces from line 5 -> 10

File: test/controllers/users_controller_test.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/controllers/users_controller_test.rb]: delete this file, it's no need

**Comment by ainulfarizah** [comment]
ready

File: config/routes.rb (hunk 5 above)
Comment (ngocvt-0484) [reply config/routes.rb:3]: wrong indent, not fix yet

File: config/routes.rb [hunk 7]
```diff
@@ -1,9 +1,10 @@
 Rails.application.routes.draw do
   scope "(:locale)", locale: /#{I18n.available_locales.join("|")}/ do
-    root 'static_pages#home'
-    get 'static_pages/help'
-    resources :users
+  root 'static_pages#home'
+  get 'static_pages/help'
```
Comment (ngocvt-0484) [inline config/routes.rb:4]: wrong indent, not fix yet

**Comment by ngocvt-0484** [comment]
please fix our comments before ready

**Comment by ainulfarizah** [comment]
image
i have already change according to comment. why at github different?

**Comment by ainulfarizah** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**